### ISSUE TITLE
Atelier Rorona Plus[BLES02050] - Glitched text

### ISSUE THREAD

**Issue by Krusher97** [RPCS3/rpcs3#4102]
Windows 7 Ultimate 64-bit SP1
Intel Core i5 6500 @ 3.20GHz
NVIDIA GeForce GTX 750 Ti

Some text in dialogue is completely blacked/glitched out. It is also glitched on the message log until you go to the next dialogue box. Happens in both OpenGL and Vulkan

https://imgur.com/a/Z8HCW

[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/1659615/RPCS3.log.gz)

**Comment by miguel-nunes** [comment]
Did you tried use the Strict Rendering Mode in the GPU tab? I think all Atelier's games use the same engine and I had the same problem with Meruru. Now the game works perfectly, no more glitches on the text.

**Comment by Krusher97** [comment]
The screenshots are with Strict Rendering Mode disabled. Enabling it causes no changes to the issue in either the Vulkan or the OpenGL renderer.

**Comment by miguel-nunes** [comment]
Strange. I had the exactly same issue with Meruru. Fixed with Strict Rendering. Did you tried use WCB?
I'm using a nVidia GPU, but I don't think this some kind of GPU issue.

**Comment by kd-11** [comment]
Afaik atelier games should have text bug fixed when strict mode is enforced. Attach log with strict mode if its not working for you.

**Comment by Krusher97** [comment]
Log with just strict rendering mode enabled.

[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/1660002/RPCS3.log.gz)

Tried it with Write Color Buffer to no avail.

**Comment by kd-11** [comment]
Try without gpu texture scaling. Strict mode should have fixed the atelier bug, if it doesn't you have no other real way around it. Not sure if rdc will help here or log with vulkan debugger enabled.

**Comment by Zangetsu38** [comment]
i don't have this bug hmm, maybe caused by regression on g8b8 hmm 
![image](https://user-images.githubusercontent.com/5261759/35345698-90a59b40-0130-11e8-8943-2400a31eba72.png)

**Comment by Zangetsu38** [comment]
@Krusher97 retry with last build

**Comment by EliEron** [comment]
I am able to reproduce this bug. I've also tried changing various options like strict rendering, WCB, GPU texture scaling and the like without that having any effect.

It's worth noting that the game being discussed here is Atelier Rorona Plus [BLES02050] and not the original release of Atelier Rorona which is what @Zangetsu38 posted above. They were released 4 years apart so I don't think they should be treated as the same game in discussions like this, as there are almost certainly some core programming differences between them.

Also I just tried with the most recent build which includes @Zangetsu38's g8b8 fix but that did not fix the issue.

My specs in case that matters:
Windows 10 64-bit
NVIDIA GeForce GTX 1080 
Intel Core i5 4670

**Comment by Krusher97** [comment]
The first time the bug occurs is when you go to the castle after the royal messenger has visited your workshop. Maybe its fixed for the original release but it certainly isn't for the Plus release.

**Comment by Krusher97** [comment]
![rpcs3_2018-03-22_00-31-07](https://user-images.githubusercontent.com/25664221/37731703-c10f04ac-2d3a-11e8-9dce-f713c06ee4b1.png)
![rpcs3_2018-03-22_00-31-13](https://user-images.githubusercontent.com/25664221/37731711-c6648382-2d3a-11e8-8f1e-9cf802c9bb8f.png)
![rpcs3_2018-03-22_00-31-20](https://user-images.githubusercontent.com/25664221/37731717-c98f195a-2d3a-11e8-9979-8b4bc99eb7e7.png)

Didn't do extensive testing but it seems to be fixed now

**Comment by OrcaXS** [comment]
It seems the problem is still there. Happens right after the screenshots of  https://github.com/RPCS3/rpcs3/issues/4102#issuecomment-375060873 though.
I've tried various options, with or without WCB/Strict render/GPU Texture Scaling enabled.

[Strict&No gpu texture scaling.log.gz](https://github.com/RPCS3/rpcs3/files/2086840/Strict.log.gz)

![rpcs3_2018-06-09_17-38-30](https://user-images.githubusercontent.com/1764123/41190220-3c6797a2-6c0d-11e8-9701-c592ba653671.png)
![rpcs3_2018-06-09_17-39-35](https://user-images.githubusercontent.com/1764123/41190328-07bda468-6c0f-11e8-8d51-ea2228528d7b.png)

**Comment by chris-l** [comment]
Please reopen the issue; is not fixed yet!

**Comment by Krusher97** [comment]
Is it a regression or does it happen in 6553 too?

**Comment by Jeneith** [comment]
it still happens in 0.0.5-6553 and latest build (0.0.5-7387) tested with both openGL and vulkan, WCB and strict rendering on looks like it's not fixed yet
![missing-texture](https://user-images.githubusercontent.com/43581503/46027944-82dc6a00-c0ef-11e8-8749-3afd118eafb2.png)
![missing-texture2](https://user-images.githubusercontent.com/43581503/46028036-c9ca5f80-c0ef-11e8-9e58-88b7e5282af8.png)
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/2416050/RPCS3.log.gz)

**Comment by Krusher97** [comment]
@kd-11 reopen pls

**Comment by Johndeep** [comment]
I wanted to upload some screenshots from Linux/Debian too, but the problem is the very same as shown by Jeneith already. What I can say is, that the problem still exists in the build v0.0.5-7402.

**Comment by kd-11** [comment]
Retest

**Comment by Krusher97** [comment]
Played for around 10 minutes and didn't encounter it

![rpcs3_2019-03-20_11-38-53](https://user-images.githubusercontent.com/25664221/54662736-0017f300-4b05-11e9-845d-a096f9cf7175.png)

Perhaps someone else can confirm this too

**Comment by Johndeep** [comment]
Played from the beginning until about after getting the first assignment (plot in the game).

Since this glitched text is really a random bug, I cannot tell, if there is more of it, but this is the only encounter of glitched textbox I had until the first assignment.
![Bildschirmfoto_2019-03-20_13-27-57](https://user-images.githubusercontent.com/306953/54685519-4bd39800-4b17-11e9-9da2-7e5f74bffc56.png)

Another random glitch encounter is this blue background, after I got to the castle and spoke to the assistance the first time. This colour bug went away with the next scene.
![Bildschirmfoto_2019-03-20_13-27-06](https://user-images.githubusercontent.com/306953/54685656-a7058a80-4b17-11e9-910f-80178dafe755.png)

Finally, even if the following bug is different from this issue, since around the #5726 commit I assume, following flickering issues of some of the textures are happening. I'm sorry for only having an abstract of the log, but I do not know why the AppImage Linux Version does not output a log file at all, neither in the main folder, nor in the .config/rpcs3. The FAQ stated that this should automatically create a log though..
```
[...]
E {RSX [0xda27d14]} RSX: Invalid texture pitch setup, width=1024, height=1024, format=0xb0(0x90) x493
E {RSX [0xda29864]} RSX: Invalid texture pitch setup, width=1024, height=1024, format=0xb0(0x90) x252
E {RSX [0xda299c4]} RSX: Invalid texture pitch setup, width=1024, height=1024, format=0xb0(0x90) x156
E {RSX [0xda29864]} RSX: Invalid texture pitch setup, width=1024, height=1024, format=0xb0(0x90) x22
[...]
E {RSX [0xda280e8]} RSX: Invalid texture pitch setup, width=1024, height=1024, format=0xb0(0x90) x118
[...]
```
Anyway, every time the textures flicker, an error in the style like above is spamming the log. To reproduce this easily, start a new game and just watch the very first scene. But it seems that there are more texture spread around the game, which also trigger this flickering.

**Comment by kd-11** [comment]
That message is only truly relevant for the developers of the game, no idea why they would set things up to be so broken. Can be clarified with hw tests and the message lowered in severity thereafter.

**Comment by digitaldude555** [comment]
It's been a year time to update.

**Comment by Johndeep** [comment]
Tested with v0.0.8-9539-11675d76_linux64.

Would really like to rerun it, but unfortunately, another bug prevents me from trying it out with Vulkan.. aka `Assertion Failed! Vulkan API call failed with unrecoverable error: Requested feature not available (VK_ERROR_FEATURE_NOT_PRESENT)`. I saw some Issues about this already.

![Bildschirmfoto_2020-02-09_17-32-52](https://user-images.githubusercontent.com/306953/74105965-b6902680-4b62-11ea-923d-0058ebbbd0ad.jpeg)
OpenGl is still broken, but it seems to have improved from last year a bit. By the way, yes, I am a proud user of a potato pc with Intel graphics, which will soon get the Intel Gallium3D driver with Mesa 20, hopefully with better performance then. But this does not change the fact, that the graphic is still glitched (with opengl).

**Comment by kd-11** [comment]
So you mean there is another bug other than the (expected) diagonal line? Where?
Diagonal line is usually not present on nvidia hardware and can usually be fixed with strict rendering mode. Also intel linux drivers aren't the best...
As for the vulkan error, try setting MSAA level to disabled. It may just work.

**Comment by Johndeep** [comment]
Oh, disabling the `Anti-Aliasing` configuration did the trick! Interesting. Thanks, the Vulkan rendering works now.
And the diagonal line is expected? Oh, okay, it seems that I missed some issues, didn't know that. ^^"

**Comment by kd-11** [comment]
Diagonal line is hardware specific and intel drivers are especially susceptible to it. You can try with "strict rendering mode" enabled, just expect not-so-great performance with it.

**Comment by Johndeep** [comment]
After so long I finally got some time to spare to check how rpcs3 is doing now.
My setup is still Debian with Intel graphics, Mesa 20.2.4 with new Iris driver. I still have to disable `Anti-Aliasing`, since MSAA is strangely still not working even with a quite new mesa driver.. but that does not matter much. I've got `V-Sync` enabled, since else the voice and rendering starts to stutter and it seems that the emulator is trapped in deadlock more often. `Strict rendering mode` is not needed anymore.

RPCS3 Version used for this bug search: 0.0.13-11380-a6a5292c (12. Dez 20)

This time, I've played the whole first part of the tutorial (until I could save again after owning the Atelier) with Japanese voice. Hopefully this will not make any differences regarding the glitches.
My impression so far, the performance is much better than before, it might be double or even more the fps than last year! There seems to be no serious rendering glitched anymore (such like wrong Shader color or plain missing models). Amazing job you guys! Probably thanks to the many improvements in rpcs3, the Mesa driver and Iris.

![screenshot-2020_12_12_17_42_50](https://user-images.githubusercontent.com/306953/101991106-c27cd580-3caa-11eb-92db-0333858f074f.jpeg)
Many bugged parts, where text are just filled with a rectangle shaped of weird colors are seemingly fixed now! Especially this part (see above picture) was prone to the glitched text, since the bubble appears "outside" the chatting environment (where background is blurred and so on)

![screenshot-2020_12_12_17_47_14](https://user-images.githubusercontent.com/306953/101991142-0c65bb80-3cab-11eb-8e2f-c91883e833be.jpeg)
Also a type of screen I've never been able to see are these "image description" dialogs, where a ready rendered image is shown as such. In the past, these images were just blanked out with black if I remember correctly..

![screenshot-2020_12_12_18_33_44](https://user-images.githubusercontent.com/306953/101991206-71211600-3cab-11eb-8ae4-06e3d1ace4e3.jpeg)
BUT as I thought the glitched text problem is finally gone, I've found this boy in an alley.. (Knuck, how dare you!)

![screenshot-2020_12_12_18_57_28_part](https://user-images.githubusercontent.com/306953/101991318-2bb11880-3cac-11eb-9d17-7d3ff4656632.jpeg)
This text should be seen here, I could make it appear again by reloading the game and meeting that boy again. It is quite frustrating to not able to debug this or log this on Linux (still no clue how to find the log). The "normal" log does not show anything.

Beside this glitch text bug (which seems to appear now less but still random), the compatibility of this game has improved a lot!

`RSX: Swapchain: Swapchain creation failed because dimensions cannot fit. Max = 1280, 720, Requested = 1920, 1080`
I noticed another problem (besides this topic issue glitch text problem). The resizing to full screen and back often does not work (if you re-scale it often enough, sometimes it works though, keyword `RSX: Swapchain`). This re-scaling problem seems to start after the intro video. It might be that the game changes something underneath while running videos and later change back to rendering. But this might be a general problem with other games too.
